### PR TITLE
Dev 3.x refactor sessions

### PR DESCRIPTION
**Pull request by uwinx**
# Description

0. Implement methods with timeout
0. Get rid of complexity creating new http-session manager

## Type of change

Please delete options that are not relevant.

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Has not been test anyhow yet.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Example:
```python
from aiogram import Bot
from aiogram.api.methods import SendMessage

async def main():
    bot = Bot("...")

    request = SendMessage(
        chat_id=124191486,
        text="text",
        request_timeout=10
    )
    print(request.request_timeout)
    await request.emit(bot)

import asyncio
asyncio.get_event_loop().run_until_complete(main())
```

### CODE REVIEW

File: tests/test_api/test_client/test_session/test_base_session.py [hunk 1]
```diff
@@ -35,13 +36,35 @@ def test_init_api(self):
...
+        session.json_loads = custom_loads
+        assert session.json_loads == custom_loads == session._json_loads
+
+        different_session = CustomSession()
+        assert all(not hasattr(different_session, attr) for attr in ("_json_loads", "_json_dumps", "_api"))
+
     def test_init_custom_api(self):
         api = TelegramAPIServer(
             base="http://example.com/{token}/{method}",
             file="http://example.com/{token}/file/{path{",
```
Comment (JrooTJunior) [inline tests/test_api/test_client/test_session/test_base_session.py:62]:
```suggestion
            file="http://example.com/{token}/file/{path}",
```

File: aiogram/api/client/session/base.py [hunk 2]
```diff
@@ -12,30 +12,41 @@
 from ..telegram import PRODUCTION, TelegramAPIServer
 
 T = TypeVar("T")
-PT = TypeVar("PT")
+_JSON_LOADS = Callable[..., Any]
+_JSON_DUMPS = Callable[..., str]
```
Comment (JrooTJunior) [inline aiogram/api/client/session/base.py:16]: Why types named in uppercase?

Comment (uwinx) [reply aiogram/api/client/session/base.py:16]: whoops caps. gotta edit JSON will remain uppercase tho. _JsonLoads should be fine, right?

Comment (JrooTJunior) [reply aiogram/api/client/session/base.py:16]: Yep

**Pull request merged**